trillian/ctfe/ct_server/main.go

```diff
@@ -331,19 +331,14 @@ func main() {
 			Certificates: []tls.Certificate{cert},
 			MinVersion:   tls.VersionTLS12,
 		}
-		srv = http.Server{
-			Addr:        *httpEndpoint,
-			Handler:     handler,
-			TLSConfig:   tlsConfig,
-			IdleTimeout: *idleConnTimeout,
-		}
+		srv = http.Server{Addr: *httpEndpoint, Handler: handler, TLSConfig: tlsConfig}
 	} else {
-		srv = http.Server{
-			Addr:        *httpEndpoint,
-			Handler:     handler,
-			IdleTimeout: *idleConnTimeout,
-		}
+		srv = http.Server{Addr: *httpEndpoint, Handler: handler}
 	}
+	if *httpIdleTimeout > 0 {
```

Comment:
A negative value for `srv.IdleTimeout` caries some meaning, see https://pkg.go.dev/net/http#Server. Did you mean to prevent this from happening? Otherwise, what's the downside to leaving the default flag value to 0 and always passing it?

Reply:
The existing behaviour doesn't set the `IdleTimeout` value. The downside of always setting this would be overriding the default `http.Server.IdleTimeout` value in case it is changed in a later version (although I don't think it's going to be changed).

Reply:
An even better way is to use the [`flag.Visit()`](https://pkg.go.dev/flag#Visit) to check whether the flag is set.

Reply:
If the current behavior is that `IdleTimeout` is not set, it inherits the 0 value, so setting the default as 0 carries forward the existing binary.

Reply:
I don't think this PR should be blocked on this, so I'll approve it and we can always change things later.

Right now, `httpIdleTimeout` default value is negative: `-1`. This negative value is used as a mechanism to branch the code, not set `IdleTimeout`, an therefore rely on its default value: `0`. I think that using negative values for such purposes is generally okay, but if the underlying variable being negative carries meaning (which is the case for `IdleTimeout` here), it increases cognitive complexity. 
Also, I've just realised that with today's implementation, if the flag is manually set to 0, the code will not explicitly set `IdleTimeout` to 0, it will just rely on its default value, which also happens to be `0`. If this default value ever changes, the flag will still be set to `0`, but `IdleTimeout` will not be anymore, which looks like a bug.

I think I prefer what you had written in the first place Bob, which I find is simpler and more explicit:
 * `httpIdleTimeout` default value is `0`, not `-1`
 * `IdleTimeout` is always set, no branching condition

> The downside of always setting this would be overriding the default http.Server.IdleTimeout value in case it is changed in a later version (although I don't think it's going to be changed).
Yes, I agree with you, that's the downside of the original approach. I don't think either that it's going to change, so I think that it's an okay compromise to make.

Reply:
Either approach should work at the moment and in the near future.

I'll go ahead to merge this PR. Thanks for the review.